tests/smoke-tests/test-staking-rewards.ts

```diff
@@ -6,23 +6,330 @@ import { ApiPromise } from "@polkadot/api";
+    if (process.env.RUN_ATSTAKE_CONSISTENCY_TESTS != "true") {
+      debug("Explicit RUN_ATSTAKE_CONSISTENCY_TESTS flag not set to 'true', skipping test");
+      this.skip();
+    }
+
+    this.timeout(TWO_HOURS);
+    this.slow(TWO_HOURS);
+    const limiter = new Bottleneck({
+      maxConcurrent: 10,
+      minTime: 100,
+    });
+    // Function to check a single Delegator's delegation to a collator
+    const checkDelegatorDelegation = async (
+      accountId: AccountId20,
+      delegatorSnapshot,
+      scheduledRequests: PalletParachainStakingDelegationRequestsScheduledRequest[]
+    ) => {
+      const { delegations: delegatorDelegations }: PalletParachainStakingDelegator = (
+        (await limiter.schedule(() =>
+          predecessorApiAt.query.parachainStaking.delegatorState(delegatorSnapshot.owner)
+        )) as any
+      ).unwrap();
+
+      const delegationAmount = delegatorDelegations.find(
+        (candidate) => candidate.owner.toString() == accountId.toString()
+      ).amount;
+
+      // Querying for pending withdrawals which affect the total
+      const scheduledRequest = scheduledRequests.find((a) => {
+        return a.delegator.toString() == delegatorSnapshot.owner.toString();
+      });
+
+      const expected =
+        scheduledRequest === undefined
+          ? delegationAmount
+          : scheduledRequest.action.isDecrease
+          ? delegationAmount.sub(scheduledRequest.action.asDecrease)
+          : scheduledRequest.action.isRevoke
+          ? delegationAmount.sub(scheduledRequest.action.asRevoke)
+          : delegationAmount;
+
+      const match = expected.eq(delegatorSnapshot.amount);
+      if (!match) {
+        debug(
+          "Snapshot amount " +
+            delegatorSnapshot.amount.toString() +
+            " does not match storage amount " +
+            delegationAmount.toString() +
+            " for delegator: " +
+            delegatorSnapshot.owner.toString() +
+            " on candidate: " +
+            accountId.toString()
+        );
       }
-      const atBlockNumber = process.env.BLOCK_NUMBER
-        ? parseInt(process.env.BLOCK_NUMBER)
-        : (await context.polkadotApi.rpc.chain.getHeader()).number.toNumber();
-      await assertRewardsAtRoundBefore(context.polkadotApi, atBlockNumber);
+      return {
+        collator: accountId.toString(),
+        delegator: delegatorSnapshot.owner.toString(),
+        match,
+      };
+    };
+    debug(`Gathering snapshot query requests for ${atStakeSnapshot.length} collators.`);
+    const promises = atStakeSnapshot.map(async (coll) => {
+      const [
+        {
+          args: [_, accountId],
+        },
+        { bond, total, delegations },
+      ] = coll;
+      const scheduledRequests = await limiter.schedule(() =>
+        predecessorApiAt.query.parachainStaking.delegationScheduledRequests(
+          accountId as AccountId20
```

Comment:
odd, is there any type erasure going on? This `as AccountId20` should be redundant here.

Reply:
Yes, this is a consequence of `atStakeSnapshot` which is calling the entries getter of the double map (with a single key). Is there a way to still import the type?

Reply:
The type is:

```typescript
import { StorageKey } from "@polkadot/types";
import { AccountId20 } from "@polkadot/types/interfaces";
import { PalletParachainStakingCollatorSnapshot } from "@polkadot/types/lookup";

let atStakeSnapshot: [StorageKey<[u32, AccountId20]>, PalletParachainStakingCollatorSnapshot][];
```

 I think there're some bugs in the test where the type erasure is causing issues. https://github.com/PureStake/moonbeam/pull/1932/files#diff-2f6aabd64c43f988835d2ef89deb70609433c387ee0e660021ceb5954bb77091R65

This code is probably trying to match the accountId from `atStakeSnapshot` but it should've been `item[0].args[1]`

Reply:
Done